components/brave_new_tab_ui/containers/newTab/settings/braveToday/useManageDirectFeeds.ts

```diff
@@ -56,43 +60,109 @@ export default function useManageDirectFeeds (publishers?: Publishers) {
   }
   const [feedInputText, setFeedInputText] = React.useState<string>()
   const [feedInputIsValid, setFeedInputIsValid] = React.useState<FeedInputValidity>(FeedInputValidity.Valid)
+  const [feedSearchResults, setFeedSearchResults] = React.useState<FeedSearchResultModel[]>([])
   const onChangeFeedInput = (e: React.ChangeEvent<HTMLInputElement>) => {
     setFeedInputText(e.target.value)
     setFeedInputIsValid(FeedInputValidity.Valid)
   }
-  const onAddSource = React.useCallback(async () => {
-    // Validate
+
+  const onSearchForSources = React.useCallback(async () => {
     if (!feedInputText) {
       return
     }
-    if (!isValidFeedUrl(feedInputText)) {
+    let feedUrlRaw = feedInputText
+    // Default protocol that should catch most cases. Make sure
+    // we check for validity after adding this prefix, and
+    // not before.
+    if (!feedUrlRaw.includes('://')) {
+      feedUrlRaw = 'https://' + feedUrlRaw
+    }
+    if (!isValidFeedUrl(feedUrlRaw)) {
       setFeedInputIsValid(FeedInputValidity.NotValid)
       return
     }
-    // Ask the backend
     setFeedInputIsValid(FeedInputValidity.Pending)
     const api = getBraveNewsAPI()
-    const result = await api.subscribeToNewDirectFeed({ url: feedInputText })
-    if (!result.isValidFeed) {
+    const { results } = await api.findFeeds({ url: feedUrlRaw })
+    if (results.length === 0) {
       setFeedInputIsValid(FeedInputValidity.NotValid)
       return
     }
-    if (result.isDuplicate) {
-      setFeedInputIsValid(FeedInputValidity.NotValid)
+    if (results.length === 1) {
+      const result = await api.subscribeToNewDirectFeed(results[0].feedUrl)
+      if (!result.isValidFeed) {
+        setFeedInputIsValid(FeedInputValidity.NotValid)
+        return
+      }
+      if (result.isDuplicate) {
+        setFeedInputIsValid(FeedInputValidity.IsDuplicate)
+        return
+      }
+      // Valid
+      setFeedInputIsValid(FeedInputValidity.Valid)
+      setFeedInputText('')
+      dispatch(todayActions.dataReceived({ publishers: result.publishers }))
       return
     }
-    // Valid
-    setFeedInputIsValid(FeedInputValidity.Valid)
-    setFeedInputText('')
+    setFeedSearchResults(results.map(r => ({ ...r, status: FeedInputValidity.Valid })))
+    setFeedInputIsValid(FeedInputValidity.HasResults)
+  }, [feedInputText, setFeedSearchResults])
```

Comment:
why is there a dep on a function expression and a one that's provided by React? i.e `setFeedSearchResults`

Reply:
addressed via https://github.com/brave/brave-core/compare/4957cf379b..f3cb1e9232